shared/utils/file-validation.ts

```diff
@@ -76,46 +73,48 @@ const validExtensions = [
 /**
  * Extracts the file extension of a given filename.
  *
- * @param filename - Name of the file
- * @return The file extension
+ * @param filename name of the file to check the extension for
+ * @return the file extension if it exists, otherwise an empty string.
  */
-export const getFileExtension = (filename: string) => path.extname(filename)
+export const getFileExtension = (filename: string): string => {
+  const splits = filename.split('.')
+  if (splits.length < 2) {
+    return ''
+  }
+  return `.${splits[splits.length - 1]}`
+}
 
 /**
- * Checks against the list of validExtensions whether the given
- * file extension is valid.
+ * Checks whether the given file extension is valid against the list of valid
+ * extensions.
  *
- * @param ext - The file extension
- * @return Whether the file extension is invalid
+ * @param ext the file extension to check
+ * @return `true` if the file extension is invalid, otherwise `false`.
  */
-export const isInvalidFileExtension = (ext: string) =>
-  !validExtensions.includes(ext.toLowerCase())
+export const isInvalidFileExtension = (ext: string): boolean =>
+  !VALID_EXTENSIONS.includes(ext.toLowerCase())
 
 /**
  * Dives into a zip file and recursively checks if it contains
  * any invalid files. A file is deemed invalid if its file extension
  * is not valid as checked by isInvalidFileExtension.
  *
- * To use this checker, first call the function and specify the platform.
- * Then use the resulting checker function to check zip files.
- *
- * @param platform - The platform the code is being
- * run on
- * @return The checker function
+ * @param dataFormat the format of the file to use in JSZip. Should be `'blob'` for `Blob` file type, and `nodebuffer` for `Buffer` file type.
+ * @param file the file to check
+ * @return array of invalid file extensions in given zip file
  */
 export const getInvalidFileExtensionsInZip = (
-  platform: FilePlatforms,
-): ((file: File | Buffer) => Promise<string[]>) => {
-  const dataFormat = platform === FilePlatforms.Browser ? 'blob' : 'nodebuffer'
-
+  dataFormat: 'blob' | 'nodebuffer',
+  file: Blob | Buffer,
+): Promise<string[]> => {
```

Comment:
in future it would be really helpful if we can keep these kinds of individual refactors separate, otherwise it takes a lot longer to figure out what's going on and whether the changes are correct

Reply:
sorry boss T_T